packages/core-api/src/apis/definitions/StorageApi.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createApiRef } from '../ApiRef';
+
+type UnsubscribeFromStore = () => void;
+type SubscribeToStoreHandler<T> = ({
+  storeName,
+  key,
+  oldValue,
+  newValue,
+}: {
+  storeName: string;
+  key: string;
+  oldValue?: T;
+  newValue?: T;
+}) => void;
+
+export type StorageApi = {
+  /**
+   * Get persistent data.
+   * @param {String} storeName Name of the store.
+   * @param {String?} key (Optional) Unique key associated with the data.
+   * If not key is specified,the whole store is returned.
+   * @param {String} key (Optional) Empty value to return if there is not data.
+   * @return {Object} data The data that should is stored.
+   */
+  getFromStore<T>(
+    storeName: string,
+    key: string,
+    defaultValue?: T,
+  ): Promise<T | undefined>;
+
+  /**
+   * Remove persistent data.
+   *
+   * @param {String} storeName Name of the store.
+   * @param {String} key Unique key associated with the data.
+   */
+  removeFromStore(storeName: string, key: string): Promise<void>;
+
+  /**
+   * Save persiswtent data.
+   *
+   * @param {String} storeName Name of the store.
+   * @param {String} key Unique key associated with the data.
+   * @param {Object} data The data that should be stored.
+   */
+  saveToStore(storeName: string, key: string, data: any): Promise<void>;
```

Comment:
Maybe we could bake remove into the save? Basically storing `undefined` is the same as removing.

Reply:
Setting as `undefined` I guess works for `localStorage` implementation, but maybe it would be different when we have different datasources? Feels a little strange to delete files on disk for instance when you save undefined? Just thinking out loud.

Reply:
Yeah kinda agree, looking for what could be removed from the API without loosing functionality 😁
We can probably keep it and reevaluate when we see some usage of the API.